source/parser.hera

```diff
@@ -80,7 +80,7 @@ IndentedApplicationAllowed
 ArgumentsWithTrailingCallExpressions
   # Since this is recursive Arguments must consume input to avoid infinite recursion
   # NOTE: Do not allow trailing template literals to match
-  Arguments ( Samedent !Backtick CallExpressionRest )*
+  Arguments ( Samedent !Backtick !ArrowFunction CallExpressionRest )*
```

Comment:
Hmm, I don't quite see why `CallExpressionRest` is what we want here. Certaintly `MemberExpressionRest` makes sense with

```coffee
foo
.bar()
```

But why do we want a call on a future same-dented line?
```coffee
foo
(bar)
```
I feel like this should compile to `foo; bar` but it currently compiles to `foo(bar)`.

So I'd propose:

```suggestion
  Arguments ( Samedent MemberExpressionRest )*
```

Oh, does this still allow the member expression to be called?

Reply:
Maybe it should be `!"("` for the lookahead since that will cover parenthesized expressions as well and arguments on the following line should be prohibited/discouraged.

Reply:
I think you are right about `MemberExpressionRest`. The additional `CallExpressions` get handled because `ArgumentsWithTrailingCallExpressions` is always either followed by a `CallExpressionRest*` or inside `CallExpressionRest`. I've got a change matching your suggestion incoming.

Reply:
Could `!Backtick !"("` be replaced by a `&"."` assertion? Equivalent?

Reply:
Oh, I see you switched to trailing `MemberExpression`; that makes me happy.

Reply:
I also added the `&"."` assertion so that shorthand `?` and `!` won't match as `MemberExpressions` when on the following line.